packages/datatrak-web/src/api/mutations/useEditUser.ts

```diff
@@ -4,45 +4,36 @@
  */
 
 import { useMutation, useQueryClient } from 'react-query';
-import { Project } from '@tupaia/types';
+import { UserAccountDetails } from '../../types';
 import { put } from '../api';
-import { Entity } from '../../types';
 
-type UserPreferences = {
-  projectId?: Project['id'];
-  countryId?: Entity['id'];
-  deleteAccountRequested?: boolean;
-};
+/**
+ * Converts a string from camel case to snake case.
+ *
+ * @remarks
+ * Ignores whitespace characters, including wordspaces and newlines. Does not handle fully-
+ * uppercase acronyms/initialisms. e.g. 'HTTPRequest' -> 'h_t_t_p_request'.
+ */
+function camelToSnakeCase(camelCaseString: string): string {
+  return camelCaseString
+    ?.split(/\.?(?=[A-Z])/)
```

Comment:
@tcaiger If I remember correctly, we copied this from somewhere. Did that source require attribution?

Reply:
to be honest I think it was stack overflow and since it's such a simple and commonly used regex I think we are all good